src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -275,11 +275,17 @@ export const createXMLHttpRequestOverride = (
 
             debug('assigned response body', this.response)
 
-            // Trigger a progress event based on the mocked response body.
-            this.trigger('progress', {
-              loaded: this.response.length,
-              total: this.response.length,
-            })
+            if (mockedResponse.body && this.response) {
+              // Presense of the mocked response implies a response body (not null).
+              // Presece of the coerced `this.response` implies the mocked body is valid.
+              const bodyBuffer = Buffer.from(mockedResponse.body)
```

Comment:
At this point, we are always creating a `Buffer` from a _string_ body. 

I don't think converting `Blob`/`ArrayBuffer` into `Buffer` once more is a good idea. Buffer representation of a string body should give a correct length.

Reply:
fine, I'm not sure about only one thing the order of the triggered events. Following the specs it seams that `loadstart` should be triggered before `onpregress`

https://xhr.spec.whatwg.org/#event-xhr-loadstart

Reply:
You're absolutely right. We can re-order the events to match the spec in a separate pull request if you don't mind.